libr/core/cmd_write.c

```diff
@@ -850,997 +844,1151 @@ static void cmd_write_pcache(RCore *core, const char *input) {
-								r_asm_code_free (ac);
-							}
-						}
-						b = a;
-						addr = nextaddr;
-					} while (a);
-					free (src);
-				} else {
-					eprintf ("Cannot open '%s'\n", file);
-				}
+		break;
+	default:
+		r_core_cmd_help (core, help_msg_wa);
+		break;
+	}
+	return 0;
+}
+
+static int wb_handler_old(void *data, const char *input) {
+	RCore *core = (RCore *)data;
+	int len = strlen (input);
+	ut8 *buf = malloc (len + 2);
+	int wseek = r_config_get_i (core->config, "cfg.wseek");
+	if (buf) {
+		len = r_hex_str2bin (input, buf);
+		if (len > 0) {
+			r_mem_copyloop (core->block, buf, core->blocksize, len);
+			if (!r_core_write_at (core, core->offset, core->block, core->blocksize)) {
+				cmd_write_fail (core);
 			} else {
-				eprintf ("Wrong argument\n");
+				WSEEK (core, core->blocksize);
 			}
-			break;
-		case 'F': // "waF"
-			if ((input[2] == ' ' || input[2] == '*')) {
-				const char *file = input + ((input[2] == '*')? 4: 3);
-				r_asm_set_pc (core->rasm, core->offset);
-				char *f = r_file_slurp (file, NULL);
-				if (f) {
-					RAsmCode *acode = r_asm_massemble (core->rasm, f);
-					if (acode) {
-						char* hex = r_asm_code_get_hex (acode);
-						if (input[2] == '*') {
-							r_cons_printf ("wx %s\n", hex);
-						} else {
-							if (r_config_get_i (core->config, "scr.prompt")) {
-								eprintf ("Written %d byte(s) (%s)=wx %s\n", acode->len, input+1, hex);
-							}
-							if (!r_core_write_at (core, core->offset, acode->bytes, acode->len)) {
-								cmd_write_fail (core);
-							} else {
-								WSEEK (core, acode->len);
-							}
-							r_core_block_read (core);
-						}
-						free (hex);
-						r_asm_code_free (acode);
-					} else {
-						eprintf ("Cannot assemble file\n");
-					}
-				} else {
-					eprintf ("Cannot slurp '%s'\n", file);
-				}
+			r_core_block_read (core);
+		} else
+			eprintf ("Wrong argument\n");
+		free (buf);
+	} else
+		eprintf ("Cannot malloc %d\n", len + 1);
+	return 0;
+}
+
+static int wm_handler_old(void *data, const char *input) {
+	RCore *core = (RCore *)data;
+	char *str = strdup (input);
+	int size = r_hex_str2bin (input, (ut8 *)str);
+	int wseek = r_config_get_i (core->config, "cfg.wseek");
+	switch (input[0]) {
+	case '\0':
+		eprintf ("Current write mask: TODO\n");
+		// TODO
+		break;
+	case '?':
+		break;
+	case '-':
+		r_io_set_write_mask (core->io, 0, 0);
+		eprintf ("Write mask disabled\n");
+		break;
+	case ' ':
+		if (size > 0) {
+			r_io_use_fd (core->io, core->file->fd);
+			r_io_set_write_mask (core->io, (const ut8 *)str, size);
+			WSEEK (core, size);
+			eprintf ("Write mask set to '");
+			int i;
+			for (i = 0; i < size; i++)
+				eprintf ("%02x", str[i]);
+			eprintf ("'\n");
+		} else
+			eprintf ("Invalid string\n");
+		break;
+	}
+	return 0;
+}
+
+static int wd_handler_old(void *data, const char *input) {
+	RCore *core = (RCore *)data;
+	if (input[0] && input[0] == ' ') {
+		char *arg, *inp = strdup (input + 1);
```

Comment:
what about using r_str_split() ?

Reply:
Sorry, I do not intend to touch existing code, it is just wasted time because all those splitting work is already done by the new parser and these functions will be removed hopefully some time in the future. I've just split the code and fixed `input+x`, that's all.